src/components/Popups/ReplyField/ReplyField.tsx

```diff
@@ -124,24 +175,82 @@ export default class ReplyField extends React.Component<Props, State> {
         setCursorPosition(editorState.getSelection().getFocusOffset());
     };
 
+    stopDefaultEvent = (event: React.SyntheticEvent): void => {
+        event.preventDefault();
+        event.stopPropagation();
+    };
+
+    setPopupListActiveItem = (index: number): void => this.setState({ activeItemIndex: index });
+
+    handleReturn = (event: React.KeyboardEvent): DraftHandleValue => {
+        const { activeItemIndex, popupReference } = this.state;
+
+        if (!popupReference) {
+            return 'not-handled';
+        }
+
+        this.stopDefaultEvent(event);
+        this.handleSelect(activeItemIndex);
+
+        return 'handled';
+    };
+
+    handleArrowHelper = (
+        event: React.KeyboardEvent,
+        getNextIndex: (activeItemIndex: number, itemsLength: number) => number,
+    ): void => {
+        const { activeItemIndex, editorState, popupReference } = this.state;
+        const collaborators = this.getCollaborators(getActiveMentionForEditorState(editorState));
+
+        if (!popupReference || !collaborators.length) {
+            return;
+        }
+
+        this.stopDefaultEvent(event);
+        const nextIndex = getNextIndex(activeItemIndex, collaborators.length);
+        this.setPopupListActiveItem(nextIndex);
+    };
+
+    handleDownArrow = (event: React.KeyboardEvent): void =>
+        this.handleArrowHelper(event, (activeItemIndex: number, itemsLength: number) =>
```

Comment:
Maybe we could have `handleArrowHelper` validate it is either up or down key event, and move the `getNextIndex` logic inline into the helper based on `isUp` or `isDown` instead of having it callback. Having these two methods call back to each other is a bit hard to read, imo -- I guess essentially move it to a single `handleArrow` method